src/briefcase/integrations/subprocess.py

```diff
@@ -88,6 +89,46 @@ def get_process_id_by_command(
     return None
 
 
+def pause_console_control_if(sub_method):
+    """Decorator for Subprocess methods to conditionally remove dynamic console
+    elements such as the Wait Bar prior to running the subprocess command.
+
+    :param sub_method: wrapped Subprocess method
+    """
+
+    @wraps(sub_method)
+    def inner(sub, cmdline, **kwargs):
+        """Evaluate whether conditions are met to remove any dynamic elements
+        in the console before returning control to Subprocess.
+
+        :param sub: Subprocess object
+        :param cmdline: list of implicit strings that will be run as subprocess command
+        :return: the return value for the Subprocess method
+        """
+        # Just run the command if no dynamic elements are active
+        if not sub.tools.input.is_console_controlled:
+            return sub_method(sub, cmdline, **kwargs)
+
+        remove_dynamic_elements = False
+
+        # Batch (.bat) scripts on Windows.
+        # If cmd.exe is interrupted with CTRL+C while running a bat script,
+        # it may prompt the user to abort the script and dynamic elements
+        # such as the Wait Bar can hide this message from the user.
+        if not remove_dynamic_elements:
+            if sub.tools.host_os == "Windows":
+                executable = str(cmdline[0]).strip() if cmdline else ""
+                remove_dynamic_elements = executable.lower().endswith(".bat")
+
+        # Run subprocess command with or without console control
+        if remove_dynamic_elements:
+            with sub.tools.input.release_console_control():
+                return sub_method(sub, cmdline, **kwargs)
+        return sub_method(sub, cmdline, **kwargs)
```

Comment:
This is a really minor stylistic thing - but I would prefer this `return` to be in an `else` clause. Functionally, there's no difference; but on a casual inspection, it wasn't immediately obvious to me that `sub_method` wasn't going to be invoked twice in the case of pre-existing dynamic elements. It won't, of course, because the `return` shortcuts the method flow - but if you don't see that *first*, it's a source of confusion. Adding an `else` clause removes that potential for confusion.

Reply:
Added the `else`.